python/src/aiconfig/editor/server/server_utils.py

```diff
@@ -0,0 +1,215 @@
+import importlib
+import importlib.util
+import logging
+import os
+import sys
+from dataclasses import dataclass
+from enum import Enum
+from types import ModuleType
+from typing import Any, Callable, NewType, Optional
+
+import lastmile_utils.lib.core.api as core_utils
+from aiconfig.Config import AIConfigRuntime
+from flask import Flask
+from pydantic import field_validator
+from result import Err, Ok, Result
+
+logging.getLogger("werkzeug").disabled = True
+
+logging.basicConfig(format=core_utils.LOGGER_FMT)
+LOGGER = logging.getLogger(__name__)
+
+log_handler = logging.FileHandler("editor_flask_server.log", mode="a")
+formatter = logging.Formatter(core_utils.LOGGER_FMT)
+log_handler.setFormatter(formatter)
+
+LOGGER.addHandler(log_handler)
+
+
+UnvalidatedPath = NewType("UnvalidatedPath", str)
+ValidatedPath = NewType("ValidatedPath", str)
+
+
+class ServerMode(Enum):
+    # debug = "DEBUG"
+    DEBUG_SERVERS = "DEBUG_SERVERS"
+    DEBUG_BACKEND = "DEBUG_BACKEND"
+    PROD = "PROD"
+
+
+class EditServerConfig(core_utils.Record):
+    server_port: int = 8080
+    aiconfig_path: Optional[str] = None
+    log_level: str | int = "INFO"
+    server_mode: ServerMode
+    parsers_module_path: str = "aiconfig_model_registry.py"
+
+    @field_validator("server_mode", mode="before")
+    def convert_to_mode(cls, value: Any) -> ServerMode:  # pylint: disable=no-self-argument
+        if isinstance(value, str):
+            try:
+                return ServerMode[value.upper()]
+            except KeyError as e:
+                raise ValueError(f"Unexpected value for mode: {value}") from e
+        return value
+
+
+@dataclass
+class ServerState:
+    aiconfig: AIConfigRuntime | None = None
+
+
+FlaskPostResponse = NewType("FlaskPostResponse", tuple[dict[str, str | core_utils.JSONObject], int])
+
+
+@dataclass(frozen=True)
+class HttpPostResponse:
+    message: str
+    aiconfig: AIConfigRuntime | None
+    code: int = 200
+
+    EXCLUDE_OPTIONS = {
+        "prompt_index": True,
+        "file_path": True,
+        "callback_manager": True,
+    }
+
+    def to_flask_format(self) -> FlaskPostResponse:
+        out: dict[str, str | core_utils.JSONObject] = {
+            "message": self.message,
+        }
+        if self.aiconfig is not None:
+            out["aiconfig"] = self.aiconfig.model_dump(exclude=HttpPostResponse.EXCLUDE_OPTIONS)
+
+        return FlaskPostResponse((out, self.code))
+
+
+def get_server_state(app: Flask) -> ServerState:
+    return app.server_state  # type: ignore
+
+
+def resolve_path(path: str) -> str:
+    return os.path.abspath(os.path.expanduser(path))
+
+
+def get_validated_path(raw_path: str, allow_create: bool = False) -> Result[ValidatedPath, str]:
+    LOGGER.debug(f"{allow_create=}")
+    if not raw_path:
+        return Err("No path provided")
```

Comment:
How do we use the aiconfig-path that's provided to the cli command? Just testing out on the client now and the client still needs to give a path to load

Reply:
Ah, yes, this is where I think it would be helpful to disambiguate load disk -> server and load server -> frontend. init_server_state runs the former, but of course you have to request the latter. I will overload /api/load as we discussed to do either 

(a) just server -> frontend (no path), or
(b) disk -> server, server -> frontend (path)